lib/airbrake/passenger.rb

```diff
@@ -0,0 +1,62 @@
+require 'set'
+require 'pathname'
+
+module Airbrake
+  # Integration for Phusion Passenger
+  module Passenger
+    IntegrationException = Class.new(::Exception)
+    class << self
+      def version
+        unless defined?(::PhusionPassenger)
+          raise IntegrationException, "PhusionPassenger not detected"
+        end
+        @version = Gem::Version.new(::PhusionPassenger::VERSION_STRING)
+      end
+
+      def version?(req)
+        reqs = req.split(',').map(&:strip).compact
+        Gem::Requirement.new(*reqs).satisfied_by?(version)
+      end
+
+      def integrations
+        return @integrations if defined?(@integrations)
+        @integrations ||= Set.new
+        dir = Pathname.new(__FILE__).dirname.join('passenger')
+        dir.each_child do |child|
+          next unless child.file? && child.readable?
+          next unless child.fnmatch?('*.rb')
+          require "airbrake/passenger/#{child.basename '.rb'}"
+        end
+        @integrations
+      end
+
+      def integration
+        return @integration if defined?(@integration)
+        @integration = integrations.to_a.detect do |i|
+          Array[i::VERSION_REQUIREMENT].any? do |vr|
+            version?(vr)
+          end
+        end
+        raise IntegrationException, "Your version of PhusionPassenger " \
+          "is not currently supported by the airbrake gem." if @integration.nil?
+        @integration
+      end
+
+      def integrate!
+        return @integrated if defined?(@integrated)
+        integration.integrate!
+      rescue IntegrationException => e
+        Kernel.warn e
+        @integrated = false
+      else
+        @integrated = true
+      end
+
+      def integrated?
+        @integrated == true
+      end
+    end
+  end
+end
```

Comment:
The convention we try to follow is to place every integration file inside its own directory. It would be nice if you could also put this file inside the passenger directory.